### PR TITLE
fix: Have RadioGroup forward aria-describedby to each of the Radio items inside

### LINKED ISSUE #27453: RadioGroup inside Field: Hint and Validation messages are not read by accessibility tools
## Repro

https://codesandbox.io/s/field-with-radiogroup-and-hint-text-ec0mc4?file=/example.tsx

## Issue

When a `RadioGroup` is inside a `Field` with `hint` and/or `validationMessage` set, the text of the hint is not read on focus of the individual radio items.

## Expected

The screen reader should announce the descriptive text when focusing on the Radio items inside the radio group.

### PR DESCRIPTION
**Pull request by behowell**
## Previous Behavior

If `aria-describedby` is set on a RadioGroup, it is not read by a screen reader because the group itself is not focusable. 

## New Behavior

Use RadioGroupContext to copy `aria-describedby` to each of the radio items inside the radio group.

## Related Issue(s)

- Fixes #27453

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/Avatar-mount_base.html">695</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/Avatar-mount.html">681</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/Button-mount_base.html">375</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/Button-mount.html">371</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/Field-mount_base.html">1258</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/Field-mount.html">1265</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/FluentProvider-mount_base.html">886</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/FluentProvider-mount.html">898</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/FluentProviderWithTheme-mount_base.html">118</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/FluentProviderWithTheme-mount.html">121</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">97</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">98</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">109</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">106</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/InfoButton-mount_base.html">15</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/InfoButton-mount.html">17</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/MakeStyles-mount_base.html">1125</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/MakeStyles-mount.html">1142</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/Persona-mount_base.html">2041</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/Persona-mount.html">1988</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/SpinButton-mount_base.html">1680</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27456/perf-test-react-components/SpinButton-mount.html">1565</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                     |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                 Change |
| :-------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr>                               |   `35.48 kB`<br />`11.126 kB` |  `35.534 kB`<br />`11.145 kB` | `54 B` <br />`19 B`  |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr>                     |   `18.889 kB`<br />`6.929 kB` |    `18.935 kB`<br />`6.94 kB` | `46 B` <br />`11 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                         | `150.829 kB`<br />`41.052 kB` | `150.883 kB`<br />`41.064 kB` | `54 B` <br />`12 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>         | `138.971 kB`<br />`35.722 kB` | `139.025 kB`<br />`35.731 kB` |  `54 B` <br />`9 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr> |    `86.8 kB`<br />`21.742 kB` |  `86.854 kB`<br />`21.751 kB` |  `54 B` <br />`9 B`  |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>           |   `86.13 kB`<br />`21.549 kB` |  `86.184 kB`<br />`21.559 kB` | `54 B` <br />`10 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `67.994 kB`<br />`18.395 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `207.512 kB`<br />`57.555 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `38.819 kB`<br />`12.351 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.446 kB`<br />`2.185 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>                                                            |  `47.119 kB`<br />`12.703 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/1071a2a382741b1328228b336c2a94de388a3963'>1071a2a382741b1328228b336c2a94de388a3963</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-radio/src/components/Radio/useRadio.tsx [hunk 1]
```diff
@@ -58,6 +59,7 @@ export const useRadio_unstable = (props: RadioProps, ref: React.Ref<HTMLInputEle
       defaultChecked,
       disabled,
       required,
+      'aria-describedby': describedByGroup,
```
Comment (khmakoto) [inline packages/react-components/react-radio/src/components/Radio/useRadio.tsx:62]: Shouldn't you merge the one that comes from the context + the one that comes from the props?

```suggestion
      'aria-describedby': `${describedByGroup} ${props["aria-describedby"]}`,
```

File: packages/react-components/react-radio/src/components/RadioGroup/RadioGroup.test.tsx [hunk 2]
```diff
@@ -87,6 +87,20 @@ describe('RadioGroup', () => {
...
+        <Radio />
+        <Radio />
+      </RadioGroup>,
+    );
+    const items = getAllByRole('radio') as HTMLInputElement[];
+    expect(items[0].getAttribute('aria-describedby')).toEqual('test-describedby');
+    expect(items[1].getAttribute('aria-describedby')).toEqual('test-describedby');
+    expect(items[2].getAttribute('aria-describedby')).toEqual('test-describedby');
+  });
+
```
Comment (khmakoto) [inline packages/react-components/react-radio/src/components/RadioGroup/RadioGroup.test.tsx:103]: If you go with my comment above, you should also add a test that sets `aria-describedby` individually to a `Radio` component aside from the one coming from the `RadioGroup` and ensure they are merged correctly.

File: packages/react-components/react-radio/src/components/Radio/useRadio.tsx (hunk 1 above)
Comment (behowell) [reply packages/react-components/react-radio/src/components/Radio/useRadio.tsx:62]: I don't think so. If someone sets `aria-describedby` on the Radio, then they are overriding the default value. They'd be responsible for including (or not including) the one from the RadioGroup.

Comment (behowell) [reply packages/react-components/react-radio/src/components/Radio/useRadio.tsx:62]: We also don't know enough about the two describedby strings to know if they can be concatenated, and what order to concatenate them.

Comment (khmakoto) [reply packages/react-components/react-radio/src/components/Radio/useRadio.tsx:62]: In that case, should this then be

```suggestion
      'aria-describedby': props['aria-describedby'] ?? describedByGroup,
```

Comment (behowell) [reply packages/react-components/react-radio/src/components/Radio/useRadio.tsx:62]: `...nativeProps.primary` is spread just below this, and will override `'aria-describedby'` if present.

Comment (behowell) [reply packages/react-components/react-radio/src/components/Radio/useRadio.tsx]: I updated ariaDescribedBy to work more like other group props (get its default value when spreading props). The result is the same, but now it works the same as the other props.

File: packages/react-components/react-radio/src/components/RadioGroup/RadioGroup.test.tsx (hunk 2 above)
Comment (behowell) [reply packages/react-components/react-radio/src/components/RadioGroup/RadioGroup.test.tsx:104]: Updated to test that aria-describedby can be overridden by an individual radio item.

**Review by sopranopillow — APPROVED** [review]

**Review by khmakoto — APPROVED** [review]

File: packages/react-components/react-radio/src/components/Radio/useRadio.tsx (hunk 1 above)
Comment (khmakoto) [reply packages/react-components/react-radio/src/components/Radio/useRadio.tsx]: Thanks Ben, resolving this convo.

**Pull request merged**